### PR TITLE
webui: setup lint for style files

### PR DESCRIPTION
**Pull request by adamkankovsky**

### CODE REVIEW

**Review by VladimirSlavik — APPROVED** [review]
Thank you, as a person who does not know this, LGTM except for cosmetics.

I won't comment on the individual rules enabled or disabled - I understand this as initial support and rule tweaking or code fixing will come later.

File: ui/webui/.stylelintrc.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "scss/at-mixin-pattern": null,
+    "scss/comment-no-empty": null,
+    "scss/dollar-variable-pattern": null,
+    "scss/double-slash-comment-whitespace-inside": null,
+    "scss/no-global-function-names": null,
+    "scss/operator-no-unspaced": null,
+    "selector-class-pattern": null,
+    "selector-id-pattern": null
+  }
+}
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/.stylelintrc.json:38]: @VladimirSlavik would be great if we could enforce this through some linter.

File: ui/webui/build.js [hunk 2]
```diff
@@ -70,6 +71,7 @@ const context = await esbuild.context({
         cleanPlugin(),
         ...lint
             ? [
+                stylelintPlugin({ filter: new RegExp(cwd + '/src/.*\\.(css?|scss?)$') }),
                 eslintPlugin({ filter: new RegExp(cwd + '\/src\/.*\.(jsx?|js?)$') })
             ]
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/build.js:76]: Hmm good question - I think now the linter who fails first blocks everything after, how the plugins are implemented internally. But also I am not 100% sure without testing it.

File: ui/webui/package.json [hunk 3]
```diff
@@ -9,7 +9,9 @@
     "watch": "ESBUILD_WATCH=true ./build.js",
     "build": "./build.js",
     "eslint": "eslint --ext .js --ext .jsx src/",
-    "eslint:fix": "eslint --fix --ext .js --ext .jsx src/"
+    "eslint:fix": "eslint --fix --ext .js --ext .jsx src/",
+    "stylelint": "stylelint 'src/**/*{.css,scss}'",
+    "stylelint:fix": "stylelint --fix 'src/**/*{.css,scss}'"
```
[the remark this answers is not in the document]
Comment (KKoukiou) [reply ui/webui/package.json:14]: +1 on this. @adamkankovsky the argument here is that keeping the trailing comma creates less noise when adding elements.

File: ui/webui/.stylelintrc.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply ui/webui/.stylelintrc.json:38]: Good idea, actually! I'll look at that.

File: ui/webui/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (adamkankovsky) [reply ui/webui/package.json:14]: Adding another element would certainly make it easier. I've always been used to deleting commas at the end, but I'm happy to adapt

File: ui/webui/build.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamkankovsky) [reply ui/webui/build.js:76]: I try it and both find the error.

[the remark this answers is not in the document]
Comment (adamkankovsky) [reply ui/webui/build.js:76]: I try it and both find the error

[the remark this answers is not in the document]
Comment (adamkankovsky) [reply ui/webui/build.js:76]: I try it and both find the error.

File: ui/webui/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (adamkankovsky) [reply ui/webui/package.json:14]: Ok, json dont allow the trailing comma. So I guess it's settled.

[the remark this answers is not in the document]
Comment (adamkankovsky) [reply ui/webui/package.json:14]: Ok, json dont allow the trailing comma. So I guess it's settled.

File: ui/webui/.stylelintrc.json [hunk 4]
```diff
@@ -35,4 +35,4 @@
     "selector-class-pattern": null,
     "selector-id-pattern": null
```
Comment (KKoukiou) [inline ui/webui/.stylelintrc.json:36]: No big deal but this was introduced in the previous commit - so it should be fixed there.

Comment (adamkankovsky) [reply ui/webui/.stylelintrc.json:36]: Done.

**Pull request merged**